docs/ota-client-guide/modules/ROOT/pages/garage-sign.adoc

```diff
@@ -0,0 +1,256 @@
+
+. As a final step, sign the new `targets.json` with your newly generated `targets` key and upload it to OTA Connect.
++
+[source,bash]
+----
+garage-sign targets sign --repo <reponame> --key-name mytargets
+garage-sign targets push --repo <reponame>
+----
+
+Please refer to xref:rotating-signing-keys.adoc#_rotate_the_keys_for_root_and_targets_metadata[rotating the keys for `root` and `targets` metadata] for additional information on generating new keys.
+
+=== Add a new software version
+
+`garage-sign` can also be used to add software packages to your repository with the command `garage-sign targets add`. The packages are downloaded by aktualizr, however `garage-sign` doesn't upload the software file. The `garage-sign targets add` commmand puts new metadata in place, hence you need to tell aktualizr where the software package can be found. This is done by uploading the file somewhere else and providing `garage-sign` with the URL of where it's available.
+
+To add a software version, perform the following steps:
+
+. First upload the software package you wish to add via a URL.
+. Next, give the URL where the software was uploaded to the command `garage-sign targets add`. This tells aktualizr where it should download the software file.
++
+[source,bash]
+----
+garage-sign targets add \
+  --repo <reponame> \
+  --format <target-format> \ <1>
+  --length <length-of-package> \ <2>
+  --name <packagename> \ <3>
+  --version <version> \ <4>
+  --sha256 <sha256sum> \ <5>
+  --hardwareids <hardware-id1> \ <6>
+  --url <url-of-software> <7>
+----
++
+<1> You must specify the format of the target image to be added--either as an ostree or binary image. In OTA Connect, the `garage-sign` tool is often used to upload `binary` images.
+<2> The length of the image to be added must be specified in bytes.
+<3> The name of the image you want to upload is specified here.
+<4> You must also specify the version number of the target image.
+<5> The sha256 hash of the software image is specified here.
+<6> This refers to the ids of hardware that your image file supports. An example can be seen in the `targets.json` file xref:customise-targets-metadata.adoc#_anatomy_of_targets_json_metadata[here] under the `custom` field.
+<7> The URL of the software package is specified here for `garage-sign` so that aktualizr will know where to find the software file to download.
+
+. Afterwards, sign your `targets.json`, which now contains your new software package, with your `targets` key and upload it to OTA Connect.
++
+[source,bash]
+----
+garage-sign targets sign --repo <reponame> --key-name mytargets
+garage-sign targets push --repo <reponame>
+----
+
+=== Define metadata expiry
+
+You can also define an expiry date for your metadata using `garage-sign` on the command line. You can do this in two ways--either specifying a fixed date and time or a specific period of time. However, only one type of expiry argument must be used to define the metadata expiry date.

```

Comment:
I don't think this is good judgement about what to include up front in the introductory paragraph. If you're going to write an introductory paragraph, it should explain the topic. Instead, you've spent two sentences explaining a minor detail about how the user can specify expiry times, and you've done it before the documentation of the tool itself.